hazelcast/src/main/java/com/hazelcast/nio/NodeIOService.java

```diff
@@ -231,7 +231,20 @@ public boolean getSocketNoDelay() {
     }
 
     @Override
-    public int getSelectorThreadCount() {
+    public int getInputSelectorThreadCount() {
+        int count = node.getGroupProperties().IO_INPUT_THREAD_COUNT.getInteger();
+        if (count != -1) {
+            return count;
+        }
+        return node.groupProperties.IO_THREAD_COUNT.getInteger();
+    }
+
+    @Override
+    public int getOutputSelectorThreadCount() {
+        int count = node.getGroupProperties().IO_OUTPUT_THREAD_COUNT.getInteger();
+        if (count != -1) {
+            return count;
+        }
         return node.groupProperties.IO_THREAD_COUNT.getInteger();
```

Comment:
same here. I know it has been like this since forever but the "boy-scout rule" should apply here.